src/d3a/models/market/market_validators.py

```diff
@@ -0,0 +1,143 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+
+from abc import ABC, abstractmethod
+from typing import Dict
+
+from d3a.models.market import Offer, Bid
+
+
+class Requirement(ABC):
+    """Define an offer/bid requirement with is_satisfied interface."""
```

Comment:
Nit: what about
```suggestion
    """Interface for offer and bid requirements."""
```

Reply:
I'm thinking about making the is_satisfied function accepts `arguments` key instead of (offer, bid,
            clearing_rate, selected_energy), what do you think?

Reply:
I think that if we don't need it, it's better to explicitly name all the arguments for the moment. If they become too many, it means that we need to create a specific class for them (e.g. `BidArguments` or something).